zcache.go

```diff
@@ -311,6 +310,53 @@ func (c *cache[K, V]) Modify(k K, f func(V) V) (V, bool) {
 	return item.Object, true
 }
 
+// Modify2 modifies the value of a key. This is similar to [Modify], but is also
+// run for keys that don't exist yet or are expired, and allows deleting keys.
+//
+// The boolean argument in the callback indicates if the key exists; for example
+// to increment only existing keys:
+//
+//	cache.Modify2("n", func(v int, ok bool) (int, bool) {
+//	    if !ok {
+//	        return 0, true
+//	    }
+//	    return v + 1, true
+//	})
+//
+// The boolean return can be used to delete keys (or not set them if they don't
+// exist). For example:
+//
+//	cache.Modify2("n", func(v int, ok bool) (int, bool) {
+//	    if !ok {
+//	        return 0, false // Don't set keys that don't exist
+//	    }
+//	    if v > 100 {
+//	        return 0, false // Delete keys higher than 100
+//	    }
+//	    return v + 1, true // Increment everything else.
+//	})
+func (c *cache[K, V]) Modify2(k K, f func(V, bool) (V, bool)) V {
```

Comment:
Wow, this was speedy, thanks @arp242! 

```suggestion
func (c *cache[K, V]) Upsert(k K, f func(V, bool) (V, bool)) V {
```

perhaps is a clear name from Database terminology?

Reply:
Also, one advantage of Modify2() is that it shows in the autocomplete for Modify():

<img width="1076" height="107" alt="cap-2025-08-06T13:46:19_border" src="https://github.com/user-attachments/assets/5d5ce78f-ac1a-4ff6-a979-16dc43f84239" />

So it's immediately obvious there is something more than Modify().

That said, Modify2() is of course an ugly name. I can't really think of anything better that starts with Modify

So idk...

Reply:
Ahh, right. Dupsert is all I've got at that point. :laughing: